general/mem_manager.cpp

```diff
@@ -87,6 +91,9 @@ static void MFEM_VERIFY_TYPES(const MemoryType h_mt, const MemoryType d_mt)
    MFEM_ASSERT(IsHostMemory(h_mt),"");
    MFEM_ASSERT(IsDeviceMemory(d_mt),"");
    const bool sync =
+      (h_mt == MemoryType::HOST && d_mt == MemoryType::DEVICE_UMPIRE) ||
+      (h_mt == MemoryType::HOST_PINNED && d_mt == MemoryType::DEVICE_UMPIRE) ||
```

Comment:
`HOST_PINNED` type could be mixed with other `device` types too, but that can be added when we'll add the equivalent tests.